httpx/middleware/recovery.go

```diff
@@ -66,11 +67,7 @@ type BasicRecovery struct {
 func (h *BasicRecovery) ServeHTTPContext(ctx context.Context, w http.ResponseWriter, r *http.Request) (err error) {
 	defer func() {
 		if v := recover(); v != nil {
-			var ok bool
-			if err, ok = v.(error); !ok {
-				err = fmt.Errorf("%v", v)
-			}
-			return
+			err = pstack.New(v)
```

Comment:
@danilotsr I think I can just ditch this whole PR and replace this line with:
``` golang
reporter.NewErrorWithContext(ctx, err, 1)
```

I'll try that